clients/logs/eks_cloudwatch_logs_client.go

```diff
@@ -0,0 +1,204 @@
+		lc.logRetentionInDays = int64(30)
+	}
+
+	flotillaMode := conf.GetString("flotilla_mode")
+	if flotillaMode != "test" {
+		sess := session.Must(session.NewSession(&aws.Config{
+			Region: aws.String(awsRegion)}))
+
+		lc.logsClient = cloudwatchlogs.New(sess)
+	}
+	lc.logger = log.New(os.Stderr, "[cloudwatchlogs] ",
+		log.Ldate|log.Ltime|log.Lshortfile)
+	lc.appLogger = appLogger
+	return lc.createNamespaceIfNotExists()
+}
+
+//
+// Logs returns all logs from the log stream identified by handle since lastSeen
+//
+func (lc *EKSCloudWatchLogsClient) Logs(definition state.Definition, run state.Run, lastSeen *string) (string, *string, error) {
+	startFromHead := true
+
+	handle := lc.toStreamName(run)
+	args := &cloudwatchlogs.GetLogEventsInput{
+		LogGroupName:  &lc.logNamespace,
+		LogStreamName: &handle,
+		StartFromHead: &startFromHead,
+	}
+
+	if lastSeen != nil && len(*lastSeen) > 0 {
+		args.NextToken = lastSeen
+	}
+
+	result, err := lc.logsClient.GetLogEvents(args)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+				// Fallback logic for legacy container names
+				if strings.HasPrefix(definition.ContainerName, definition.GroupName) {
+					definition.ContainerName = strings.Replace(
+						definition.ContainerName, fmt.Sprintf("%s-", definition.GroupName), "", -1)
+					return lc.Logs(definition, run, lastSeen)
+				}
+
+				return "", nil, exceptions.MissingResource{err.Error()}
+			} else if request.IsErrorThrottle(err) {
+				lc.logger.Printf(
+					"thottled getting logs; definition_id: %s, run_id: %s, error: %+v\n",
+					definition.DefinitionID, run.RunID, err)
+				return "", lastSeen, nil
+			}
+		}
+		return "", nil, errors.Wrap(err, "problem getting logs")
+	}
+
+	if len(result.Events) == 0 {
+		return "", result.NextForwardToken, nil
+	}
+
+	message := lc.logsToMessage(result.Events)
+	return message, result.NextForwardToken, nil
+}
+
+func (lc *EKSCloudWatchLogsClient) toStreamName(run state.Run) string {
+	return fmt.Sprintf("%s", run.PodName)
+}
+
+func (lc *EKSCloudWatchLogsClient) logsToMessage(events []*cloudwatchlogs.OutputLogEvent) string {
+	sort.Sort(byTimestamp(events))
+
+	messages := make([]string, len(events))
+	for i, event := range events {
+		var l EKSCloudWatchLog
+		err := json.Unmarshal([]byte(*event.Message), &l)
+		if err != nil {
+			messages[i] = *event.Message
+		}
+		messages[i] = l.Log
+	}
+	return strings.Join(messages, "")
+}
+
+func (lc *EKSCloudWatchLogsClient) createNamespaceIfNotExists() error {
+	exists, err := lc.namespaceExists()
+	if err != nil {
+		return errors.Wrapf(err, "problem checking if log namespace [%s] exists", lc.logNamespace)
+	}
+	if !exists {
+		return lc.createNamespace()
+	}
+	return nil
+}
+
+func (lc *EKSCloudWatchLogsClient) namespaceExists() (bool, error) {
+	result, err := lc.logsClient.DescribeLogGroups(&cloudwatchlogs.DescribeLogGroupsInput{
+		LogGroupNamePrefix: &lc.logNamespace,
```

Comment:
This behavior probably isn't what we want with EKS, the will be a single namespace. Maybe we dont need the Prefixing?